lib/squadster/formations/services/update_squad.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Squadster.Formations.Services.UpdateSquad do
+  import SquadsterWeb.Gettext
+
+  alias Squadster.Repo
+  alias Squadster.Formations.Squad
+
+  def call(args, squad) do
+    changeset = squad |> Squad.changeset(args)
+
+    case changeset |> Repo.update do
+      {:ok, squad} ->
+        changeset.changes |> notify_students(squad)
+        {:ok, squad}
+      {:error, reason} -> {:error, reason}
+    end
+  end
+
+  defp notify_students(changes, squad) do
+    Squadster.Accounts.Tasks.Notify.start_link([
+      message: text(changes),
+      target: squad,
+      options: [
+        skip_commander: true,
+        skip_deputy: true,
+        skip_journalist: true
+      ]
+    ])
+  end
+
+  defp text(changes) do
+    gettext("Squad info has been changed:") <> class_day(changes) <> squad_number(changes) <> advertisment(changes)
+  end
+
+  defp class_day(changes) do
+    if changes[:class_day] do
```

Comment:
The same approach is used in these functions. Maybe make sense to separate it somehow, if it possible.
```
def message(scope, value) do
  "\n\n" <> gettext("New squad #{scope}: %{#{scope}}", scope: value)
end
```
Something like that
Then, we can do like this: 
`if changes[some_scope], do: message(some_scope, changes[some_scope]), else: ""`

Reply:
Have no idea how to fix it, I can get rid of `if` use here by passing scope as a param, but will need another condition for text itself, so it'll be even harder to read